### ISSUE TITLE
Can I append changelog?

### ISSUE THREAD

**Issue by staehler** [mooltiverse/nyx#183]
I hope I haven't overlooked anything in the documentation. How can I grow the changelog so that all versions remain in the CHANGELOG.md. It currently looks as if the changelog is emptied and rewritten with every version change. I would like to have the complete change history in my CHANGELOG.md.
Is this possible?
I run nyx in a gitlab CI/CD.
And thank you very much for this really cool tool :-)

**Comment by staehler** [comment]
Thank you very much for your quick response.
I run the versioning only on one branch (DevOps), which is the main branch. From this branch the code is deployed on pre-production and later (after some test period) on production environment.
Do you think, there is a a chance to get this feature?

**Comment by staehler** [comment]
On the other side, if there is no huge benefit for your software with this feature, I'll find a workaround. E.g. cat old and new changelog together within my CI/CD. No big deal, I hope ;-)

**Comment by flelli** [comment]
You're right, I can implement it, just outlining the limitation that the commit history is limited to the current branch backward, and let the user decide.
I'm putting it in the backlog.

**Comment by staehler** [comment]
Thanks a lot!

Meanwhile and if somebody interested in, I've found following solution:
In my CI/CD I my CHANGELOG.md to .changelog.old and run `nyx mark`, where the regarding .nyx.json locks like:
```
{
  "changelog":{
    "path":"CHANGELOG.md",
    "template": "changelog.tpl",
    "sections":{
      "Added":"^feat$",
      "Fixed":"^fix$",
      "CI/CD":"^ci$"
    },
 ```
and the template (derived from [here](https://raw.githubusercontent.com/mooltiverse/nyx/main/modules/java/main/src/main/resources/changelog.tpl))
```
# Changelog

{{#releases}}
## {{name}} ({{date}})

{{#sections}}
### {{name}}

{{#commits}}
* [[{{#short5}}{{sha}}{{/short5}}]({{#environmentVariable}}CI_PROJECT_URL{{/environmentVariable}}/-/commit/{{sha}})] {{message.fullMessage}} ({{authorAction.identity.name}}, {{#timestampISO8601}}{{commitAction.timeStamp.timeStamp}}{{/timestampISO8601}})

{{/commits}}
{{^commits}}
No changes.
{{/commits}}
{{/sections}}
{{^sections}}
No changes.
{{/sections}}
{{/releases}}
{{^releases}}
No releases.
{{/releases}}

{{#fileContent}}.changelog.old{{/fileContent}}
```

There are some variables used from gittlab CI environment, which can/should be substituted.

**Comment by flelli** [comment]
Hi @staehler ,

these features are on their way. They will be available in preview in [2.5.0-alpha.3](https://github.com/mooltiverse/nyx/releases/tag/2.5.0-alpha.3) ([still building](https://github.com/mooltiverse/nyx/actions/runs/5975697044) as I'm writing but I need to leave the keyboard so it's there for your reference) and they will be generally available in the upcoming [2.5.0](https://github.com/mooltiverse/nyx/releases/tag/2.5.0).

You now have the new [`append`](https://mooltiverse.github.io/nyx/guide/user/configuration-reference/changelog/#append) configuration option inside the changelog configuration (the link will work once the final version is published). When not defined it will override the output file (also for backward compatibility), but if you set it to `head` or `tail` the output will be inserted on top of the file or appended at the end, respectively, so you can control whether to have most recent versions on top (reverse order) or at the end (natural order).

**Comment by flelli** [comment]
Release [2.5.0](https://github.com/mooltiverse/nyx/releases/tag/2.5.0) is out. Changelog is [here](https://github.com/mooltiverse/nyx/blob/main/RELEASE_NOTES.md#250).
